### PR TITLE
[OSD-20514] Retry send SL notification

### PR DESCRIPTION
**Pull request by a7vicky**
### What type of PR is this?
_(feature)_

### What this PR does / why we need it?
Retry if it fails to send SL.

### Which Jira/Github issue(s) this PR fixes?

_Fixes #_[OSD-20514](https://issues.redhat.com/browse/OSD-20514)

### Pre-checks (if applicable):
- [x] Tested latest changes against a cluster
- [ ] Ran `make generate` command locally to validate code changes
- [ ] Included documentation changes with PR

### CODE REVIEW

File: pkg/handlers/webhookreceiver.go [hunk 1]
```diff
@@ -146,17 +145,34 @@ func (h *WebhookReceiverHandler) processAlert(alert template.Alert, mnl *oav1alp
 		// This is not an error state
 		return nil
 	}
+
+	var attempts int = 5
+	var sleep time.Duration = 30
```
Comment (tkong-redhat) [inline pkg/handlers/webhookreceiver.go:150]: Was Duration unit is nanoseconds? Just wonder if 30 nanoseconds is longer enough for retry

Comment (a7vicky) [reply pkg/handlers/webhookreceiver.go]: Thanks @tkong-redhat for the review. I have updated this in my latest commit PTAL

File: pkg/httpchecker/reattempt.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	"time"
+)
+
+func init() {
+	rand.New(rand.NewSource(time.Now().UnixNano())) //nolint:gosec
+}
+
+type stop struct {
+	error
+}
```
Comment (devppratik) [inline pkg/httpchecker/reattempt.go:14]: Does this need to a struct separately if we only have error here?

File: pkg/httpchecker/urlavailabilitychecker.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+func (c *UrlHTTPChecker) UrlAvailabilityCheck(url string) error {
+	resp, err := c.Client.Get(url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode >= 200 && resp.StatusCode < 300 {
+		return nil
+	}
```
Comment (devppratik) [inline pkg/httpchecker/urlavailabilitychecker.go:39]: Can we improve this check?
Maybe using somethine like 
```
if resp.StatusCode >= http.StatusOK && resp.StatusCode < http.StatusMultipleChoices {
    return nil
}
```
OR 
```
if resp.StatusCode/100 == 2 {
    return nil
}
```

**Review by devppratik — COMMENTED** [review]
LGTM on the code. Just a few non blocking review

File: pkg/httpchecker/urlavailabilitychecker.go (hunk 3 above)
Comment (a7vicky) [reply pkg/httpchecker/urlavailabilitychecker.go]: Thanks @devppratik for the review. I have updated in my latest commit

File: pkg/httpchecker/reattempt.go (hunk 2 above)
Comment (a7vicky) [reply pkg/httpchecker/reattempt.go:16]: Yes this is not needed but I have keep it consistent as same and ref from MUO repo

**Pull request merged**